### PR TITLE
P1224R0 C++ Standard Library Issues to be moved in San Diego except 2697, 2960 and 3134

### LINKED ISSUE #2410: [2018-11 LWG Motion 4] P1224R0 C++ Standard Library Issues to be moved in San Diego except...
except 2697, 2960 and 3134

### PR DESCRIPTION
**Pull request by burblebee**
Fixes #2410.

### CODE REVIEW

**Comment by jensmaurer** [comment]
@burblebee, there is a Travis CI failure showing overfull \hboxes here.

File: source/iostreams.tex [hunk 1]
```diff
@@ -7965,7 +7965,7 @@
 Constructs an object of class
 \tcode{basic_istringstream<charT, traits>},
 initializing the base class with
-\tcode{basic_istream<charT, traits>(\&sb)}\iref{istream}
+\tcode{basic_istream<charT, traits>(addressof(sb))}\iref{istream}
 and initializing \tcode{sb} with
 \tcode{basic_string\-buf<charT, traits, Allocator>(str, which | ios_base::in)}\iref{stringbuf.cons}.
 \end{itemdescr}
```
Comment (timsong-cpp) [inline source/iostreams.tex]: This is missing the `basic_istream<charT, traits>::set_rdbuf(\&sb)`on line 7983 below.

File: source/iostreams.tex [hunk 2]
```diff
@@ -8152,7 +8152,7 @@
 Constructs an object of class
 \tcode{basic_ostringstream<charT, traits>},
 initializing the base class with
-\tcode{basic_ostream<charT, traits>(\&sb)}\iref{ostream}
+\tcode{basic_ostream<charT, traits>(addressof(sb))}\iref{ostream}
 and initializing \tcode{sb} with
 \tcode{basic_string\-buf<charT, traits, Allocator>(str, which | ios_base::out)}\iref{stringbuf.cons}.
 \end{itemdescr}
```
Comment (timsong-cpp) [inline source/iostreams.tex]: And likewise here, it missed the `basic_ostream<charT, traits>::set_rdbuf(\&sb)` on line 8170.

**Comment by tkoeppe** [comment]
@zygoloid: beware of a stray space in the commit message of "LWG2936 Path comparison is defined in terms of the generic format" (near the end); something to be fixed during rebasing?

File: source/iostreams.tex [hunk 3]
```diff
@@ -8756,6 +8756,9 @@
 
 \begin{itemdescr}
 \pnum
+\expects
+\tcode{s} shall point to a NTCTS\iref{defns.ntcts}.
+\pnum
```
Comment (tkoeppe) [inline source/iostreams.tex]: We should have a newline above the `\pnum` here.

File: source/iostreams.tex [hunk 4]
```diff
@@ -8805,9 +8806,9 @@
 If the open operation succeeds and
 \tcode{(mode \& ios_base::ate) != 0},
 positions the file to the end
-(as if by calling
-\tcode{fseek(file, 0, SEEK_END)}).\footnote{The macro
-\tcode{SEEK_END}
+(as if by calling \tcode{fseek(file, 0, SEEK_END)}, where
+\tcode{file} is the pointer returned by calling \tcode{fopen}).
```
Comment (tkoeppe) [inline source/iostreams.tex]: Escape the space at the end of the line; there must be no whitespace before a footnote.

**Comment by tkoeppe** [comment]
@zygoloid: beware of the commit message for "LWG3025 Map-like container deduction guides should use pair, not pair" missing the two template argument lists after both instances of "pair"

File: source/algorithms.tex [hunk 5]
```diff
@@ -120,13 +120,16 @@
...
-construct
+as its iterator argument with value type \tcode{T},
+it should work correctly in the construct
 \tcode{pred(*first)} contextually converted to \tcode{bool}\iref{conv}.
 The function object
 \tcode{pred}
 shall not apply any non-constant
 function through the dereferenced iterator.
+Given a glvalue \tcode{u} of type (possibly const) \tcode{T} that
+designates the same object as \tcode{*first, pred(u)}
```
Comment (tkoeppe) [inline source/algorithms.tex]: The `tcode` needs to be split and the comma needs to be a text comma

File: source/algorithms.tex [hunk 6]
```diff
@@ -155,6 +159,20 @@
 construct \tcode{binary_pred(*first1, value)} contextually converted to \tcode{bool}\iref{conv}.
 \tcode{binary_pred} shall not
 apply any non-constant function through the dereferenced iterators.
+HERE
```
Comment (tkoeppe) [inline source/algorithms.tex]: Stray debug text?

File: source/utilities.tex [hunk 7]
```diff
@@ -12617,12 +12617,13 @@
...
-\tcode{\placeholdernc{OUTERMOST}(x.outer_allocator())}
-otherwise;
+\tcode{\placeholdernc{OUTERMOST}(x)} is
+\tcode{\placeholdernc{OUTERMOST}(x.outer_allocator())} if
+the expression \tcode{x.outer_allocator()} is
+valid~\iref{temp.deduct} and
+\tcode{x} otherwise;
 \tcode{\placeholdernc{OUTERMOST_ALLOC_TRAITS}(x)} is
-\tcode{allocator_traits<decltype(\placeholdernc{OUTERMOST}(x))>}.
+\tcode{allocator_traits<remove_reference_t<decltype(\placeholdernc{OUTERMOST}(x)))>}.
```
Comment (tkoeppe) [inline source/utilities.tex]: The close should have an angled bracket (`>`), not a round parenthesis (`)`).

File: source/lib-intro.tex [hunk 8]
```diff
@@ -2580,9 +2580,11 @@
...
 \indextext{unit!translation}%
 A translation unit shall not \tcode{\#define} or \tcode{\#undef}
-names lexically identical to keywords, to the identifiers listed in
-\tref{identifiers.special}, or to the \grammarterm{attribute-token}{s} described
-in~\ref{dcl.attr},
+names lexically identical
+to keywords,
+to the identifiers listed in \tref{identifiers.special},
+to the \grammarterm{attribute-token}{s} described in~\ref{dcl.attr}, or
+to the identifiers expects or ensures,
```
Comment (tkoeppe) [inline source/lib-intro.tex]: code font for the two identifiers

File: source/time.tex [hunk 9]
```diff
@@ -3323,46 +3323,36 @@
...
-    using period                    = ratio<@\unspecnc@, @\unspec@>;
-    using duration                  = chrono::duration<rep, period>;
-    using time_point                = chrono::time_point<file_clock>;
-    static constexpr bool is_steady = @\unspec@;
-
-    static time_point now() noexcept;
-
-    // Conversion functions, see below
-  };
+  using file_clock = @\seebelow@>;
```
Comment (tkoeppe) [inline source/time.tex]: stray angle bracket

File: source/algorithms.tex (hunk 6 above)
Comment (burblebee) [reply source/algorithms.tex]: OOPS!  yup.  :)

**Comment by burblebee** [comment]
@tkoeppe Thanks for your review!  And thanks to @zygoloid for fixing all the problems you found! 
It shows how important it is to make sure each motion is carefully reviewed by one of us - I think we need a better process to make sure that happens.

**Pull request merged**